### PR TITLE
Add named logging with LOGNAME to OMPL interface

### PR DESCRIPTION
**Pull request by JeroenDM**
### Description

As is done in other parts of MoveIt #1079 #1179, this pull request adds named logging to all source files of the `moveit_planners_ompl` package. (For some files with a single log statement, it might be overkill, but it is consistent...)

**Important** I'm not confident whether it is ok to put this outside a namespace. But in most source files of the OMPL interface, all code is put outside of a namespace and all names are prepended with `ompl_interface::`. Maybe this is worth another cleanup? Or is there a reason for this?

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by JeroenDM** [comment]
I think I used the wrong clang-format configurations, I will fix it.

**Comment by JeroenDM** [comment]
Tavis reports failing tests in `moveit/moveit_ros/perception/mesh_filter/test/mesh_filter_test`. I don't know what to make of it.

**Comment by rhaschke** [comment]
The failing mesh_filter tests are unrelated (see #2212 and a pending fix in #2180).

**Comment by JeroenDM** [comment]
Just curious, is there another way to quickly check which tests failed on Travis, besides reading through the XML output?

**Comment by v4hn** [comment]
The short summaries are listed at the bottom of the log: https://travis-ci.com/github/ros-planning/moveit/jobs/362068846#L1837

**Comment by JeroenDM** [comment]
Aaah, I looked over it, sorry. Thank you :)

File: moveit_planners/ompl/ompl_interface/src/detail/constrained_goal_sampler.cpp [hunk 1]
```diff
@@ -41,6 +41,8 @@
 
 #include <utility>
 
+static const std::string LOGNAME{ "constrained_goal_sampler" };
```
Comment (mamoll) [inline moveit_planners/ompl/ompl_interface/src/detail/constrained_goal_sampler.cpp:44]: Please put this inside the ompl_interface namespace (the same applies to all cases below). In addition, write it as
```
constexpr char LOGNAME[] = "constrained_goal_sampler";
```
Constexprs offer the compilers more options for optimization. A std::string cannot be created via constexpr.

Comment (JeroenDM) [reply moveit_planners/ompl/ompl_interface/src/detail/constrained_goal_sampler.cpp:44]: Ok!

Can I put all class method definitions inside the namespace (as is done in many other parts of MoveIt, but not [here](https://github.com/ros-planning/moveit/blob/877e1d51560f701b60b79e04a852ec104ad77de0/moveit_planners/ompl/ompl_interface/src/planning_context_manager.cpp#L88) for example). Or is the preferred way to do `ompl_interface::LOGNAME` in all the log statements?

**Comment by v4hn** [comment]
> Or is the preferred way to do `ompl_interface::LOGNAME` in all the log statements?

LOGNAME should not be used with a namespace definition. If you do, I guess it would mean you're lying. :-)

**Comment by JeroenDM** [comment]
I must admit I don't really understand what I did.

From what I understand (after some googling) the constexpr adheres to normal namespace behavior in this context. The LOGNAME's in the rest of the file work because they are used inside class methods that are defined inside the same namespace, and therefore LOGNAME is available.

Ideally, I would like to extend the namespace scope to the end of the source files, as done in [ompl_planner_manager.cpp](https://github.com/ros-planning/moveit/blob/84cce2ae2a342abd180b4ff97063c3c6dc8b7aa8/moveit_planners/ompl/ompl_interface/src/ompl_planner_manager.cpp#L313). It would make the code less verbose. But that's probably worth a separate issue / pull request.

**Pull request merged**